File: app/src/main/java/com/lumiyaviewer/lumiya/res/executors/ResourceCleanupExecutor.java
package com.lumiyaviewer.lumiya.res.executors;

import java.util.concurrent.ScheduledThreadPoolExecutor;
import java.util.concurrent.ThreadFactory;

public class ResourceCleanupExecutor extends ScheduledThreadPoolExecutor {

    private static class InstanceHolder {
        /* access modifiers changed from: private */
        public static final ResourceCleanupExecutor Instance = new ResourceCleanupExecutor();

        private InstanceHolder() {
        }
    }

    public ResourceCleanupExecutor() {
        super(1, new ThreadFactory() {
            public Thread newThread(Runnable runnable) {
                return new Thread(runnable, "ResourceCleanup");
            }
    }

    public static ResourceCleanupExecutor getInstance() {
        return InstanceHolder.Instance;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/res/executors/ResourceCleanupExecutor.kt
package com.lumiyaviewer.lumiya.res.executors

import java.util.concurrent.ScheduledThreadPoolExecutor
import java.util.concurrent.ThreadFactory

/**
 * Singleton executor used by [com.lumiyaviewer.lumiya.res.ResourceManager] to
 * periodically prune weak caches and cancel orphaned requests.
 */
class ResourceCleanupExecutor private constructor() :
    ScheduledThreadPoolExecutor(1, ThreadFactory { task ->
        Thread(task, "ResourceCleanup").apply { isDaemon = true }
    }) {

    companion object {
        private val instance = ResourceCleanupExecutor()

        @JvmStatic
        fun getInstance(): ResourceCleanupExecutor = instance
    }
}
